File: app/src/main/java/com/faciee/cti/valbastrelu/eticket/ui/common/SectionsPagerAdapter.java
package com.faciee.cti.valbastrelu.eticket.ui.common;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

/**
 * A {@link FragmentPagerAdapter} that returns a fragment corresponding to
 * one of the sections/tabs/pages.
 */
public class SectionsPagerAdapter extends FragmentPagerAdapter {
	
	private final List<Fragment> mFragmentList = new ArrayList<>();
	private final List<String> mFragmentTitleList = new ArrayList<>();
	
	public SectionsPagerAdapter(FragmentManager fm) {
		super(fm);
	}
	
	public void addFragment(Fragment fragment, String title){
		mFragmentList.add(fragment);
		mFragmentTitleList.add(title);
	}
	
	@Override
	public CharSequence getPageTitle(int position) {
		return mFragmentTitleList.get(position);
	}
	
	@Override
	public Fragment getItem(int position) {
		// getItem is called to instantiate the fragment for the given page.
		// Return a PlaceholderFragment (defined as a static inner class below).
		return mFragmentList.get(position);
	}
	
	@Override
	public int getCount() {
		return mFragmentList.size();
	}
}

File: app/src/main/java/com/faciee/cti/valbastrelu/eticket/ui/common/adapters/SectionsPagerAdapter.kt
package com.faciee.cti.valbastrelu.eticket.ui.common.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import java.util.*


class SectionsPagerAdapter(val fragmentManager: FragmentManager) : FragmentPagerAdapter(fragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
	private val mFragmentList: MutableList<Fragment> = ArrayList()
	private val mFragmentTitleList: MutableList<String> = ArrayList()

	override fun getPageTitle(position: Int): CharSequence = mFragmentTitleList[position]

	override fun getItem(position: Int): Fragment = mFragmentList[position]

	override fun getCount(): Int = mFragmentList.size

	fun makeFragmentName(containerViewId: Int, id: Long): String {
		return "android:switcher:$containerViewId:$id"
	}

	fun getFragmentForPosition(viewPagerId: Int, position: Int): Fragment? {
		val tag = makeFragmentName(viewPagerId, getItemId(position))
		return fragmentManager.findFragmentByTag(tag)
	}

	fun addFragment(fragment: Fragment, title: String) {
		mFragmentList.add(fragment)
		mFragmentTitleList.add(title)
	}
}